aiida/orm/nodes/data/dict.py

```diff
@@ -19,7 +19,25 @@
 
 
 class Dict(Data):
-    """`Data` sub class to represent a dictionary."""
+    """`Data` sub class to represent a dictionary.
+
+    The dictionary key-value pairs are stored in the database in a column named `attributes`.
+    We will then say that these are "database-attributes", or "attributes in a database sense".
+    However, these are not "python-attributes" ("attributes in the python sense"), and can only
```

Comment:
Although I see the point of adding this, given that this was not clear for you either, this does not just apply to the `Dict` nodes, but to all nodes. All node specific data are stored as attributes in the database. So although this may help clear things up, it may also confuse, since it may suggest that this is behavior specific to the `Dict` plugin.

I would propose a text something like:
```
    The dictionary contents of a `Dict` node are stored in the database as attributes. The dictionary can be initialized
    through the `dict` argument in the constructor. After construction, values can be retrieved and updated through the
    item getters and setters, respectively:

        node['key'] = 'value'

    Alternatively, the `dict` property returns an instance of the `AttributeManager` that can be used to get and set
    values through attribute notation:

        node.dict.key = 'value'

    Note that trying to set dictionary values directly on the node, e.g. `node.key = value`, will not work as intended.
    It will merely set the `key` attribute on the node instance, but will not be stored in the database. As soon as the
    node goes out of scope, the value will be lost.

    Finally, all dictionary mutations will be forbidden once the node is stored.
```

Reply:
Ok, would it still be ok to add something like this between the last two paragraphs?

```
    node goes out of scope, the value will be lost.

    Note that here it becomes apparent the difference of something being an "attribute of a node"
    (in the sense that it is stored in the "attribute" column of the database when the node is stored)
    and something being an "attribute of a python object" (in the sense of being able to modify and
    access it as if it was a property of the variable).

    Finally, all dictionary mutations will be forbidden once the node is stored.
```